src/protocol/prf_sharding/mod.rs

```diff
@@ -123,6 +131,93 @@ where
     rows_chunked_by_user
 }
 
+pub async fn attribution_and_capping_and_aggregation<C, BK, TV, F>(
+    sh_ctx: C,
+    input_rows: Vec<PrfShardedIpaInputRow<BK, TV>>,
+    num_saturating_sum_bits: usize,
+) -> Result<Vec<Replicated<F>>, Error>
+where
+    C: UpgradableContext,
+    C::UpgradedContext<Gf2>: UpgradedContext<Gf2, Share = Replicated<Gf2>>,
+    BK: GaloisField,
+    TV: GaloisField,
+    F: Field,
+{
+    // Call attribution_and_capping
+    let result: Vec<CappedAttributionOutputs> =
+        attribution_and_capping(sh_ctx, input_rows, num_saturating_sum_bits).await?;
+
+    let validator = &sh_ctx.validator::<Fp32BitPrime>();
```

Comment:
Is this correct? Since everything is in prime field we should need validator in Fp32BitPrime field also?